web/components/contract/contract-tabs.tsx

```diff
@@ -364,54 +365,66 @@ export const CommentsTabContent = memo(function CommentsTabContent(props: {
           }
         />
       )}
-      {parentComments.slice(0, parentCommentsToRender).map((parent) => (
-        <FeedCommentThread
-          key={parent.id}
-          contract={contract}
-          parentComment={parent}
-          threadComments={commentsByParent[parent.id] ?? []}
-          trackingLocation={'contract page'}
-          idInUrl={idToHighlight}
-          showReplies={
-            !isBountiedQuestion || (!!user && user.id === contract.creatorId)
-          }
-          childrenBountyTotal={
-            contract.outcomeType == 'BOUNTIED_QUESTION'
-              ? childrensBounties[parent.id]
-              : undefined
-          }
-          onSubmitReply={loadNewer}
-          bets={bets?.filter(
-            (b) =>
-              b.replyToCommentId &&
-              [parent]
-                .concat(commentsByParent[parent.id] ?? [])
-                .map((c) => c.id)
-                .includes(b.replyToCommentId)
-          )}
-        />
-      ))}
 
-      <div className="relative w-full">
-        <VisibilityObserver
-          onVisibilityUpdated={onVisibilityUpdated}
-          className="pointer-events-none absolute bottom-0 h-[75vh]"
-        />
-      </div>
-
-      {!user && (
-        <Button
-          onClick={withTracking(
-            firebaseLogin,
-            'sign up to comment button click'
-          )}
-          className={clsx('mt-4', comments.length > 0 && 'ml-12')}
-          size="lg"
-          color="gradient"
-        >
-          Sign up to comment <ArrowRightIcon className="ml-2 h-4 w-4" />
-        </Button>
-      )}
+      {parentComments
+        .filter((comment) => comment.pinned)
+        .map((parent) => (
+          <FeedCommentThread
+            key={parent.id}
+            contract={contract}
+            parentComment={parent}
+            threadComments={commentsByParent[parent.id] ?? []}
+            trackingLocation={'contract page'}
+            idInUrl={idToHighlight}
+            showReplies={
+              !isBountiedQuestion || (!!user && user.id === contract.creatorId)
+            }
+            childrenBountyTotal={
+              contract.outcomeType == 'BOUNTIED_QUESTION'
+                ? childrensBounties[parent.id]
+                : undefined
+            }
+            onSubmitReply={loadNewer}
+            bets={bets?.filter(
+              (b) =>
+                b.replyToCommentId &&
+                [parent]
+                  .concat(commentsByParent[parent.id] ?? [])
+                  .map((c) => c.id)
+                  .includes(b.replyToCommentId)
+            )}
+          />
```

Comment:
Duplicating this code for the feed thread is not ideal.

You can simply insert the pinned comments first in the parent comments array. Ask chat gpt

Reply:
Before seeing this I had already made some changes. I'm not inserting them into the parent comments array, but after reviewing my new code let me know if I should still make this change.